python/ray/rllib/models/preprocessors.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+
+class Preprocessor(object):
+    """Defines an abstract observation preprocessor function."""
+
+    def transform_shape(self, obs_shape):
+        """Returns the preprocessed observation shape."""
+        raise NotImplementedError
+
+    def transform(self, observation):
+        """Returns the preprocessed observation."""
+        raise NotImplementedError
+
+
+class AtariPixelPreprocessor(Preprocessor):
+    def transform_shape(self, obs_shape):
+        return (80, 80, 3)
+
+    # TODO(ekl) why does this need to return an extra size-1 dim (the [None])
+    def transform(self, observation):
+        """Downsamples images from (210, 160, 3) to (80, 80, 3)."""
+        return (observation[25:-25:2, ::2, :][None] - 128) / 128
```

Comment:
@pcmoritz do you know why the `[None]` is needed here? This causes the shape to be `(1, 80, 80, 3)` which trips up ES unless you squeeze out the extra dim.

However PPO seems to be happy with the extra dim, and in fact it broke when I tried getting rid of it:

```
Traceback (most recent call last):
  File "/home/eric/Desktop/ray-private/python/ray/worker.py", line 732, in _process_task
    self.actors[task.actor_id().id()], *arguments)
  File "/home/eric/Desktop/ray-private/python/ray/rllib/policy_gradient/agent.py", line 194, in compute_steps
    for key, val in trajectory.items()}
  File "/home/eric/Desktop/ray-private/python/ray/rllib/policy_gradient/agent.py", line 194, in <dictcomp>
    for key, val in trajectory.items()}
IndexError: boolean index did not match indexed array along dimension 0; dimension is 87120 but corresponding boolean dimension is 1089
```

Didn't investigate further.